promises.cf

```diff
@@ -208,9 +207,20 @@ bundle common cfengine_stdlib
       };
 
     !cfengine_3_5::
-      # CFEngine 3.6 and higher can include through a secondary file
+      # CFEngine 3.6 can include through a secondary file
+      # CFEngine version 3.6 and prior use the split library to avoid syntax
+      # errors introduced by new functionality. For example new functions.
+      # This also works for 3.8 because local_libdir should be set to lib
+      # instead of lib/3.8
       "inputs" slist => { "$(sys.local_libdir)/stdlib.cf" };
 
+    cfengine_3_7::
+      # CFEngine 3.7 has local_libdir set to $(sys.inputdir)/lib/3.7, but with
+      # the @if macro support we can re-unify the split library for 3.7+ so we
+      # specify the unified lib instead of local_libdir.
+      "inputs" slist => { "lib/stdlib.cf" };
```

Comment:
This won't always work.  Instead, I would use `$(sys.local_libdir)/..` with the understanding that it's a temporary workaround.